src/template.yml

```diff
@@ -1287,7 +1287,8 @@ Resources:
                 - aws s3 sync ./adf-build/shared s3://$DEPLOYMENT_ACCOUNT_BUCKET/adf-build --quiet
                 # Base templates:
                 - aws s3 sync . s3://$S3_BUCKET --quiet --delete
-                - aws s3 sync ./adf-accounts s3://$ACCOUNT_BUCKET --quiet
+                # Upload account files to the ACCOUNT_BUCKET
+                - python adf-build/shared/helpers/sync_to_s3.py --extension .yml --upload-with-metadata execution_id=${CODEBUILD_BUILD_NUMBER} --recursive adf-accounts s3://$ACCOUNT_BUCKET
```

Comment:
I don't think this is the right env variable here.

Reply:
I think we want to add the execution ID from the CodePipeline to it via env variables and then use that

Reply:
Although you can query the execution Id using the pipeline state query. Just depends if we want those additional api calls and IAM permissions

Reply:
I looked into this, it is possible to get the CodePipeline execution id.

Unfortunately [CodeBuild environment variables](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html) do not expose the CodePipeline execution id as an environment variable just yet.

Apparently you can use the [`#{codepipeline.PipelineExecutionId}` variable in the CodePipeline definition](https://docs.aws.amazon.com/codepipeline/latest/userguide/reference-variables.html#reference-variables-workflow).
We would need to update the CodePipeline definitions for this to be supported.
I would need to test how that would merge with setting environment variables on the CodeBuild project itself.

I am wondering though if this will update if you retry a single stage. Lets say CodeBuild fails due to another failure, will it get a new ID if you only retry that stage?

Reply:
It looks like adding the variable from the pipeline won't impact project level variables: 
<img width="1368" alt="Screenshot 2022-09-16 at 09 17 48" src="https://user-images.githubusercontent.com/7794788/190591541-162d1583-4598-40a2-bd07-1f966a222867.png">
See the lowercase codepipe_execution_id variable. 

I've tested single stage retry and the execution Id is the same, which is the desired outcome?

Reply:
Thanks for testing that. The number is a bit long though, but I guess we can use that.

I guess rerunning with the same ID would not hurt.
It would only use/apply the ID when there are content changes any way.

I will add the execution id to this PR, thanks!

Reply:
Done.